File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/service/model/question/Question.java
package uk.co.claritysoftware.exam.slammr.webapp.service.model.question;

import java.time.ZonedDateTime;
import java.util.List;
import java.util.Set;

import lombok.Builder;
import lombok.Value;

/**
 * Pojo defining an Exam Slammr Question
 */
@Value
@Builder
public class Question {

	private String id;

	private String summary;

	private String questionText;

	private List<AnswerOption> answers;

	private Set<String> tags;

	private Set<String> certifications;

	private Set<FurtherReading> furtherReadings;

	private String createdBy;

	private ZonedDateTime createdDateTime;

	private String updatedBy;

	private ZonedDateTime updatedDateTime;

	private QuestionStatus status;

	private int votes;

}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/service/model/question/Question.kt
package uk.co.claritysoftware.exam.slammr.webapp.service.model.question

import java.time.ZonedDateTime

/**
 * Pojo defining an Exam Slammr Question
 */
data class Question(val id: String?,
					val slug: String?,
					val summary: String,
					val questionText: String,
					val answers: Set<AnswerOption>,
					val tags: Set<String>,
					val certifications: Set<String>,
					val furtherReadings: Set<FurtherReading>,
					val createdBy: String,
					val createdDateTime: ZonedDateTime,
					val updatedBy: String?,
					val updatedDateTime: ZonedDateTime?,
					val status: QuestionStatus,
					val votes: Int)